ormin/queries.nim

```diff
@@ -982,8 +982,10 @@ proc addFields(n: NimNode; b: ProtoBuilder): NimNode =
     expectKind x, nnkRecList
     doAssert b.retType.len == b.retNames.len, "ormin: types and column names do not match"
     for i in 0 ..< b.retType.len:
-      x.add newTree(nnkIdentDefs, newTree(nnkPostfix, ident"*", ident(b.retNames[i])),
-                    b.retType[i][1], newEmptyNode())
+      var typ = b.retType[i][1]
+      if typ.kind == nnkIdent and typ.strVal == "string":
+        typ = ident("kstring")
```

Comment:
No. kstring is Karax specific, has no place in Ormin.

Reply:
Can i map type directly:
```
protocol "chatclient.nim":
  common:
    when defined(js):
      type string = cstring
```
If can, let user decide how to map the types. Ormin does not need to make any assumptions

Reply:
Or another solution:
```
protocol "chatclient.nim":
  typemap:
    type kstring = string
  common:
    when defined(js):
      type kstring = cstring
```
Add typemap description like common, replace the user defined type according to the declaration: `type kstring = string`, replace string type to kstring, generated code like:
```
when defined(js):
  type
    kstring = cstring
type
  TextMessage* = ref object
    content*: kstring 
    creation*: kstring 
    author*: int
    name*: kstring 
```
Which one is better？

Reply:
``typemap``? What's wrong with the original way of dealing with ``kstring``, the way I wrote it? It was ideal and required no Ormin specific code.